Road/source/features/data_cleaning.py

```diff
@@ -0,0 +1,185 @@
+# import needed libraries
+import pandas as pd 
+import numpy as np 
+import re
+import string 
+
+# Local imports
+# from source.constants import PUNCT_TO_REMOVE
+import sys
+import os
+target = os.path.abspath(__file__)
+while(target.split("\\")[-1]!="Road"):
+    target = os.path.dirname(target)
+sys.path.append(target) 
+
+# Local imports
+from source.constants import PUNCT_TO_REMOVE
+
+# For arabic diacritics 
+from pyarabic.araby import strip_tashkeel, is_tashkeel, is_arabicrange
+
+# For converting emojis
+import emoji
+# pip install emot
+import emot
+
+
+def remove_diacritics(text: str) -> str:
+    """
+    Remove diacritics from Arabic text.
+
+    Args:
+        text (str): The input Arabic text.
+
+    Returns:
+        str: The Arabic text with diacritics removed.
+    """
+    # initialize an empty string to hold the text without diacritics
+    text_without_diacritics = ""
+    for char in text:
+        # check if the character is in the Arabic Unicode range
+        if is_arabicrange(char):
+            # check if the character is a diacritic
+            if is_tashkeel(char):
+                # if it is a diacritic, remove it using the strip_tashkeel function
+                text_without_diacritics += strip_tashkeel(char)
+            else:
+                # if it is not a diacritic, append it to the text without diacritics
+                text_without_diacritics += char
+        else:
+            # if the character is not in the Arabic Unicode range, append it to the text without diacritics
+            text_without_diacritics += char
+    
+    return text_without_diacritics
+
+def remove_emojis(text: str) -> str:
+    """
+    Remove emojis from text.
+
+    Args:
+        text (str): The input text.
+
+    Returns:
+        str: The input text with emojis and emoticons removed.
+    """
+    new_text = ''
+    for character in text:
+        if not emoji.is_emoji(character):
+            new_text += character
+    return new_text
+
+def remove_urls(text: str) -> str:
+    """
+    Remove URLs from text.
+
+    Args:
+        text (str): The input text with URLs.
+
+    Returns:
+        str: The input text with URLs removed.
+    """
+    url_pattern = re.compile(r'https?://\S+|www\.\S+')
+    return url_pattern.sub(r'', text)
+
+def remove_phone_numbers(text: str) -> str:
+    """
+    Remove phone numbers from text.
+
+    Args:
+        text (str): The input text with phone numbers.
+
+    Returns:
+        str: The input text with phone numbers removed.
+    """
+    text = re.sub(r'(\+?\d{2,4}[ -]?)?\d{9,10}', '', text)
+    return text
```

Comment:
You can return the text immediately instead of save the result in the `text` variable then return it.